h/services/user.py

```diff
@@ -56,7 +62,55 @@ def fetch(self, userid_or_username, authority=None):
             username = parts['username']
             authority = parts['domain']
 
-        return self._cached_fetch(username, authority)
+        # The cache is keyed by (username, authority) tuples.
+        cache_key = (username, authority)
+
+        if cache_key not in self._cache:
+            self._cache[cache_key] = (self.session.query(User)
+                                      .filter_by(username=username)
+                                      .filter_by(authority=authority)
+                                      .one_or_none())
+
+        return self._cache[cache_key]
+
+    def fetch_all(self, userids):
+        """
+        Fetch a list of users by their userids.
+
+        This function fetches users based on the list, adds them to the internal
+        cache and then returns the list of users. This is especially useful
+        when needing to access multiple user objects without loading them one-by-one.
+
+        It will only attempt to load the users that aren't already cached.
```

Comment:
This docstring could be a little clearer about what happens if a userid is not found.

Reply:
Fixed.